### PR TITLE
Updating group.removeMember api to send an email for rescinded party invite

### LINKED ISSUE #7618: Need notification when party invite rescinded
When a party invitation is sent, a user with that option enabled gets an email. If the user is removed from the party, there's an email for that as well. There is, however, no email sent for if the invitation is rescinded before you join -- making it possible to get an email and sign in to accept, but find nothing. 

This email should probably not be exactly the same as the removed from party email (it was the same in APIv2), but it should exist.

### PR DESCRIPTION
**Pull request by Hus274**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #7618 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
This change makes it so a user is sent an email notification when they have an invite rescinded for a group or when they are removed from the group even when the user doesn't send a message. This still requires text to be added to the email server by an admin.

Let me know any feedback you have!

### CODE REVIEW

File: test/api/v3/integration/groups/POST-groups_id_removeMember.test.js [hunk 1]
```diff
@@ -80,6 +89,20 @@ describe('POST /groups/:groupId/removeMember/:memberId', () => {
...
+      expect(email.sendTxn).to.be.calledOnce;
+      expect(email.sendTxn.args[0][1]).to.be.eql('guild-invite-rescinded');
+    });
+
+    it('sends email to removed user', async () => {
+      await leader.post(`/groups/${guild._id}/removeMember/${member._id}`);
+
+      expect(email.sendTxn).to.be.calledOnce;
+      expect(email.sendTxn.args[0][1]).to.be.eql('kicked-from-guild');
+    });
```
Comment (khipkin) [inline test/api/v3/integration/groups/POST-groups_id_removeMember.test.js:107]: Suggestion: for all of these tests, make sure the email is sent to the correct user (the one who got removed or had their invite rescinded).

Comment (paglias) [reply test/api/v3/integration/groups/POST-groups_id_removeMember.test.js:107]: 👍

Comment (Hus274) [reply test/api/v3/integration/groups/POST-groups_id_removeMember.test.js:107]: Good catch, not sure why I didn't do this in the first place.

**Comment by Hus274** [comment]
Added test to validate the email was being sent to the right user.

**Comment by Hus274** [comment]
Fixed linting issue. Should be good to go now.

**Comment by TheHollidayInn** [comment]
@paglias Do we need to create this email or has it already been created?

**Comment by paglias** [comment]
waiting for the text, then I'll create the email and merge

**Comment by Hus274** [comment]
@palgias would you like me to submit some text?

**Comment by paglias** [comment]
This is the text we're using when you're kicked from a party

```
You have been removed from your Party, "*|GROUP_NAME|*". You can form your own party
under Social > Party, or look for a new one in the <a href="*|BASE_URL|**|PARTY_WANTED_GUILD|*">Party Wanted Guild</a>.

*|IF:MESSAGE|*
    <br><br>

    Your former Party Leader has provided the following message: <br>
    <blockquote>*|MESSAGE|*</blockquote>
*|END:IF|*
```

what about this for this new case?

```
Your invitation for the Party "*|GROUP_NAME|*" has been rescinded. You can form your own party
under Social > Party, or look for a new one in the <a href="*|BASE_URL|**|PARTY_WANTED_GUILD|*">Party Wanted Guild</a>.

*|IF:MESSAGE|*
    <br><br>

    The Party Leader has provided the following message: <br>
    <blockquote>*|MESSAGE|*</blockquote>
*|END:IF|*
```

**Comment by Hus274** [comment]
This is the text @lemoness suggested in the [issue](https://github.com/HabitRPG/habitica/issues/7618). 

> Yes, this makes sense. Perhaps it should have the same subject line so that it could thread?
>
> As for the body of the email, let's do:
>
> "Your Party invitation has been rescinded by the party leader, so your invitation link will no longer work. To find a different Party, try posting in the Party Wanted Guild. Good luck!"

I think it looks good to me but it might be worth adding the note that the invitation link will no longer work.

**Comment by paglias** [comment]
Ohh I didn't notice that text from @lemoness , never mind then, we can use that

**Comment by Hus274** [comment]
Sounds good!

**Comment by paglias** [comment]
@Hus274 I've created the emails, so we're good to go!

**Pull request merged**